C/common/config_category.cpp

```diff
@@ -872,6 +872,50 @@ vector<string> ConfigCategory::getOptions(const string& name) const
 	throw new ConfigItemNotFound();
 }
 
+/**
+ * Return the permissions of the configuration category item
+ *
+ * @param name	The name of the configuration item to return
+ * @return vector<string>	The configuration item permissions
+ * @throws exception if the item does not exist in the category
+ */
+vector<string> ConfigCategory::getPermissions(const string& name) const
+{
+	for (unsigned int i = 0; i < m_items.size(); i++)
+	{
+		if (name.compare(m_items[i]->m_name) == 0)
+		{
+			return m_items[i]->m_permissions;
+		}
+	}
+	throw new ConfigItemNotFound();
+}
+
```

Comment:
As you have added getter for this, Don't we require setter for the same item?

Reply:
Nothing is using the currently, setting is an issue with knowing how has permissions. It is a construction time only thing to set it currently. We can tackle that if and when we need it.

Reply:
Okay, I thought it was required in `setItemAttribute`. Seems like only required for an optional attribute in categoryItem constructor and its JSON representation function i.e `toJSON` and `defaultToJSON`